Tests/GRPCTests/GRPCReflectionServiceTests/GRPCReflectionServiceTests.swift

```diff
@@ -0,0 +1,362 @@
+    let reflectionService = try ReflectionService(
+      fileDescriptorProtos: self.makeProtosWithDependencies()
+    )
+
+    let server = try Server.insecure(group: MultiThreadedEventLoopGroup.singleton)
+      .withServiceProviders([reflectionService])
+      .withLogger(self.serverLogger)
+      .bind(host: "127.0.0.1", port: 0)
+      .wait()
+    self.server = server
+
+    let channel = try GRPCChannelPool.with(
+      target: .hostAndPort("127.0.0.1", server.channel.localAddress!.port!),
+      transportSecurity: .plaintext,
+      eventLoopGroup: MultiThreadedEventLoopGroup.singleton
+    ) {
+      $0.backgroundActivityLogger = self.clientLogger
+    }
+    self.channel = channel
+  }
+
+  override func tearDown() {
+    if let channel = self.channel {
+      XCTAssertNoThrow(try channel.close().wait())
+    }
+    if let server = self.server {
+      XCTAssertNoThrow(try server.close().wait())
+    }
+
+    super.tearDown()
+  }
+
+  func testFileByFileName() async throws {
+    try self.setUpServerAndChannel()
+    let client = Reflection_ServerReflectionAsyncClient(channel: self.channel!)
+    let serviceReflectionInfo = client.makeServerReflectionInfoCall()
+    try await serviceReflectionInfo.requestStream.send(
+      .with {
+        $0.host = "127.0.0.1"
+        $0.fileByFilename = "bar1.proto"
+      }
+    )
+    serviceReflectionInfo.requestStream.finish()
+
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    let message = try await iterator.next()
+    let receivedFileDescriptorProto =
+      try Google_Protobuf_FileDescriptorProto(
+        serializedData: (message?.fileDescriptorResponse
+          .fileDescriptorProto[0])!
+      )
+    XCTAssertEqual(receivedFileDescriptorProto.name, "bar1.proto")
+    XCTAssertEqual(
+      receivedFileDescriptorProto.service.count,
+      1
+    )
+    XCTAssertEqual(
+      receivedFileDescriptorProto.service.first!.method.first!.name,
+      "testMethod1"
+    )
+    XCTAssertEqual(message?.fileDescriptorResponse.fileDescriptorProto.count, 4)
+  }
+
+  func testListServices() async throws {
+    try self.setUpServerAndChannel()
+    let client = Reflection_ServerReflectionAsyncClient(channel: self.channel!)
+    let serviceReflectionInfo = client.makeServerReflectionInfoCall()
+
+    try await serviceReflectionInfo.requestStream.send(
+      .with {
+        $0.host = "127.0.0.1"
+        $0.listServices = "services"
+      }
+    )
+
+    serviceReflectionInfo.requestStream.finish()
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    let message = try await iterator.next()
+    let receivedServices = message?.listServicesResponse.service.map { $0.name }
+    XCTAssertEqual(receivedServices?.count, 4)
+
+    let servicesNames = self.getServicesNamesFromProtos(protos: self.makeProtosWithDependencies())
+    for serviceName in receivedServices! {
+      XCTAssertTrue(servicesNames.contains(serviceName))
+    }
```

Comment:
It would be better to compare the sorted names of services: if `XCTAssertTrue` fails then the message will just say it was false but expected true. We lose the information about the expected and actual service names.

If we compare the equality of two sorted arrays and they're different we can see what's missing.